kotlin/samples/tictactoe/common/src/main/java/com/squareup/sample/gameworkflow/RunGameWorkflow.kt

```diff
@@ -0,0 +1,233 @@
+import com.squareup.sample.gameworkflow.RunGameState.Playing
+import com.squareup.sample.gameworkflow.SyncState.SAVED
+import com.squareup.sample.gameworkflow.SyncState.SAVE_FAILED
+import com.squareup.sample.gameworkflow.SyncState.SAVING
+import com.squareup.sample.panel.PanelContainerScreen
+import com.squareup.sample.panel.asPanelOver
+import com.squareup.viewregistry.AlertContainerScreen
+import com.squareup.viewregistry.AlertScreen
+import com.squareup.viewregistry.AlertScreen.Button.NEGATIVE
+import com.squareup.viewregistry.AlertScreen.Button.NEUTRAL
+import com.squareup.viewregistry.AlertScreen.Button.POSITIVE
+import com.squareup.viewregistry.AlertScreen.Event.ButtonClicked
+import com.squareup.viewregistry.AlertScreen.Event.Canceled
+import com.squareup.workflow.EventHandler
+import com.squareup.workflow.Snapshot
+import com.squareup.workflow.Workflow
+import com.squareup.workflow.WorkflowAction.Companion.emitOutput
+import com.squareup.workflow.WorkflowAction.Companion.enterState
+import com.squareup.workflow.WorkflowContext
+import com.squareup.workflow.invoke
+import com.squareup.workflow.rx2.onSuccess
+
+enum class RunGameResult {
+  CanceledStart,
+  FinishedPlaying
+}
+
+/**
+ * We define this otherwise redundant interface to keep composite reactors
+ * that build on [RunGameWorkflow] decoupled from it, for ease of testing.
+ */
+interface RunGameWorkflow :
+    Workflow<Unit, RunGameState, RunGameResult, AlertContainerScreen<PanelContainerScreen<*, *>>>
+
+/**
+ * Runs the screens around a Tic Tac Toe game: prompts for player names, runs a
+ * confirm quit screen, and offers a chance to play again. Delegates to [TakeTurnsWorkflow]
+ * for the actual playing of the game.
+ */
+class RealRunGameWorkflow(
+  private val takeTurnsWorkflow: TakeTurnsWorkflow,
+  private val gameLog: GameLog
+) : RunGameWorkflow {
+
+  override fun initialState(input: Unit): RunGameState = NewGame()
+
+  override fun compose(
+    input: Unit,
+    state: RunGameState,
+    context: WorkflowContext<RunGameState, RunGameResult>
+  ): AlertContainerScreen<PanelContainerScreen<*, *>> {
+    return when (state) {
+      is NewGame -> {
+        val emptyGameScreen = GamePlayScreen()
+
+        val eventHandler = context.onEvent<NewGameScreen.Event> { event ->
+          when (event) {
+            CancelNewGame -> emitOutput(CanceledStart)
+            is StartGame -> enterState(Playing(PlayerInfo(event.x, event.o)))
+          }
+        }
+
+        subflowScreen(
+            base = emptyGameScreen,
+            subflow = NewGameScreen(state.defaultXName, state.defaultOName, eventHandler)
+        )
+      }
+
+      is Playing -> {
+        // context.compose starts takeTurnsWorkflow, or keeps it running if it was
+        // already going. TakeTurnsWorkflow.compose is immediately called,
+        // and the GamePlayScreen it renders is immediately returned.
+        val takeTurnsScreen = context.compose(takeTurnsWorkflow, state.playerInfo) {
```

Comment:
Nit: Might be more clear, as a sample, to explicitly name the lambda parameter `output` here.